src/main/scala/org/camunda/feel/FeelEngine.scala

```diff
@@ -196,7 +202,12 @@ class FeelEngine(
   }
 
   def eval(exp: ParsedExpression, context: Context): EvalExpressionResult =
-    validate(exp).flatMap(_ => eval(exp, rootContext + context))
+    Try {
+      validate(exp).flatMap(_ => eval(exp, rootContext + context))
+    }.recover(failure =>
+        Left(
+          Failure(s"failed to evaluate expression '${exp.text}' : $failure")))
+      .get
```

Comment:
Should like 214 also be wrapped inside a Try?

Reply:
It is possible but not necessary. 

The method

```
def eval(exp: ParsedExpression, variables: Map[String, Any] = Map()): EvalExpressionResult
```

delegates to

```
def eval(exp: ParsedExpression, context: Context): EvalExpressionResult
```

So, the evaluation is already wrapped in a `Try`.